Veda Charitha - Job Specific Template Changes

Charitha2009 opened the pull request:
# Description
<img width="549" height="273" alt="Screenshot 2026-03-12 at 11 14 37 PM" src="https://github.com/user-attachments/assets/c12b8d29-8716-43ab-a577-231f8b2b467a" />

## Related PRS (if any):
This frontend PR is related to the development backend PR.

## Main changes explained:

- Job-aware application form: Full fields + engineering-only questions, basic validation/toasts, and a “Learn more” link that opens the modal.
- Refactored job modal: Full-width “Seeking Experienced {Job Title}” → One Community intro → description with auto-built sections; keeps image grid + external details link; close (×) inside card.
- Polish & consistency: Helpers normalize ad data; updated CSS (stacked header, chips, spacing, responsive/dark-mode); consistent “Select a position” placeholder and immediate form on selection.

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. Clear site data/cache
4. log as admin user
5. go to http://localhost:5173/collaboration/analytics
6. Select a job From dropdown: Verify the ApplicationForm renders with additional fields and, for engineering and other roles.
7. Learn more link: At the bottom of the form, click Learn more about this position → modal opens.
Confirm: If it opens job description modal in brief.

## Screenshots or videos of changes:
<img width="1460" height="726" alt="Screenshot 2025-09-05 at 8 09 34 PM" src="https://github.com/user-attachments/assets/79623945-991e-488c-a14a-cafbd608c21e" />
<img width="1172" height="685" alt="Screenshot 2025-09-05 at 8 09 49 PM" src="https://github.com/user-attachments/assets/bf29bee3-c0af-4a03-a40c-b531d0e476ec" />
<img width="1089" height="584" alt="Screenshot 2025-09-05 at 8 10 05 PM" src="https://github.com/user-attachments/assets/6327e184-69d5-4685-968e-4b4b80d3c50b" />
<img width="217" height="470" alt="Screenshot 2025-09-05 at 8 10 26 PM" src="https://github.com/user-attachments/assets/5b419469-2768-435c-8e5a-b70b2a85f488" />

Charitha2009 committed:
added specific page template based on role

Charitha2009 committed:
fixed responsiveness for mobile view

hemanthvenkat commented:
Verified this PR by selecting a job from the dropdown and confirming the ApplicationForm renders with additional fields for engineering and other roles.
Checked the Learn more link at the bottom, which opened the job description modal.
All functionality worked as expected without issues.

https://github.com/user-attachments/assets/52ed7783-a25a-4949-91d4-6bd18ddc755c

SudheeshTD reviewed (dismissed):
The Job Specific template page displays all the additional fields, and it shows job description at the end of the page and all the other details properly renders.
<img width="1933" height="1306" alt="image" src="https://github.com/user-attachments/assets/bcd4d7da-950b-448f-8e08-2d0ca7c04777" />
<img width="1914" height="1191" alt="image" src="https://github.com/user-attachments/assets/0cd5f233-a3ae-4e08-bb45-e5e2c301ec63" />
<img width="1665" height="1191" alt="image" src="https://github.com/user-attachments/assets/ff3a71b3-257b-456f-8ab5-38ac0baad5c9" />


<img width="1736" height="1226" alt="image" src="https://github.com/user-attachments/assets/8b39a0d5-ec3c-4a98-bcae-7cc164b3516c" />

akshith312 reviewed (dismissed):
Tested the PR locally. Logged in as Admin user.
Forms are working as expected and are responsive as well.

<img width="1918" height="967" alt="image" src="https://github.com/user-attachments/assets/73292465-02fa-4a66-98b3-9d66340efadb" />
<img width="1917" height="963" alt="image" src="https://github.com/user-attachments/assets/ba70e3c6-def0-4ca6-83f7-b6b51b747a3f" />
<img width="868" height="912" alt="image" src="https://github.com/user-attachments/assets/1e4664c9-29ac-40cd-913a-014e23ad4440" />
<img width="826" height="902" alt="image" src="https://github.com/user-attachments/assets/7654ca7a-c50c-4728-aa08-cf0aef5adced" />

Prem203 reviewed (changes requested):
Checked the PR with an admin account. The form renders correctly when a job is selected. The card pops up correctly with brief description when clicked on the link.

However in dark mode, the form itself is dark but the background still remains white. The colors in dark mode for title and questions makes it difficult to read, while the numbers are not visible.

<img width="1652" height="967" alt="image" src="https://github.com/user-attachments/assets/897fcb88-381d-4ab8-9512-8fa1918e5b25" />

<img width="1331" height="840" alt="image" src="https://github.com/user-attachments/assets/3a58c863-d0c6-4af4-8eac-ca0ffc5f3c37" />

<img width="1344" height="864" alt="image" src="https://github.com/user-attachments/assets/0dd417d7-9cfd-42e6-a0a9-81fa6c26734c" />

<img width="1852" height="962" alt="image" src="https://github.com/user-attachments/assets/ad1a7c4e-23a6-4d18-97ed-e2bccd0018e6" />

Swetha-1306 reviewed (approved):
Tested the PR. Everything works as described including the link for specific job description depending on the job user chooses.
<img width="1470" height="879" alt="Pasted Graphic 1" src="https://github.com/user-attachments/assets/e6e4cabe-75fb-470a-9968-c4b3d43b2343" />
<img width="1470" height="879" alt="Pasted Graphic 2" src="https://github.com/user-attachments/assets/24e068ce-55d8-4175-b9d5-2f6d3e708758" />
<img width="1470" height="879" alt="Seeking Experienced Experienced MERN Stack" src="https://github.com/user-attachments/assets/44e0a117-8139-4225-b6ef-1eb43357b48b" />

harshavarma29 reviewed (changes requested):
I have reviewed the PR using the Admin account, and the job filters are working as expected. However, I noticed the following issues:

1) Dark mode issue: The dark theme does not cover the entire UI, and some UI elements are not changing properly in dark mode.
    <img width="1888" height="861" alt="dark_mode_issue" src="https://github.com/user-attachments/assets/6c7e875e-bcd6-4820-9297-3fbfdbdc7048" />

2) Job cards in the "Show Summaries" tab do not have a fixed size:
    https://github.com/user-attachments/assets/6f01d2c3-938a-45c1-a1bc-df8c1770f1bd

3) Page responsiveness issue: The page is not responsive between 640px and 1030px:
    https://github.com/user-attachments/assets/198437e7-77bc-4505-8668-dbd9086a3a8f

Anusha-Gali reviewed (changes requested):
Hi Charitha,

I have reviewed your PR locally and have identified the below issues.
1. When we click back, the texts used to search for the job should be cleared out
2. The dark mode is not enabled for the text in the form as well as modal popup

<img width="1276" height="741" alt="Screenshot 2025-11-21 at 8 22 38 PM" src="https://github.com/user-attachments/assets/44404e8a-1081-417d-a6a1-4cfbffd8eadd" />
<img width="1276" height="741" alt="Screenshot 2025-11-21 at 8 22 52 PM" src="https://github.com/user-attachments/assets/bd64481b-4744-4e17-9635-b07627eb09b0" />
<img width="1276" height="741" alt="Screenshot 2025-11-21 at 8 23 19 PM" src="https://github.com/user-attachments/assets/4fc5fff2-234a-46f8-8edf-04cfc1f80c4e" />
<img width="1276" height="741" alt="Screenshot 2025-11-21 at 8 23 22 PM" src="https://github.com/user-attachments/assets/69497bf4-2a8e-4dd8-a3a0-40c561d394d3" />
**Issue 1: When we click back, the texts used to search for the job should be cleared out**
<img width="1276" height="741" alt="Screenshot 2025-11-21 at 8 23 45 PM" src="https://github.com/user-attachments/assets/f612f27d-f0e6-457f-8f96-862a2b86239a" />
<img width="1276" height="741" alt="Screenshot 2025-11-21 at 8 23 59 PM" src="https://github.com/user-attachments/assets/fbc13dea-af24-402f-ac53-a83ae1a10c4d" />
**Issue 2: The dark mode is not enabled for the text in the form as well as modal popup**
<img width="1276" height="741" alt="Screenshot 2025-11-21 at 8 24 26 PM" src="https://github.com/user-attachments/assets/2a760416-3b85-4015-9797-7b22062c0361" />
<img width="1276" height="741" alt="Screenshot 2025-11-21 at 8 24 32 PM" src="https://github.com/user-attachments/assets/66344004-7eb2-4dc3-921f-4e23c3cda1e1" />

rohanrastogi311 reviewed (changes requested):
Hi Veda, well done with the implementations. One big issue is the job description isn't showing properly (check image below) 

<img width="1618" height="1032" alt="PR 4026 Screenshot " src="https://github.com/user-attachments/assets/8ea68fdd-c8a6-47d3-9425-4af687a32ec6" />
<img width="1627" height="1037" alt="PR 4026 Screenshot 2" src="https://github.com/user-attachments/assets/881d6db3-0877-43d8-8840-dc3babee942d" />
<img width="1724" height="818" alt="PR 4026 Screenshot 3" src="https://github.com/user-attachments/assets/522bbb7c-ae2b-4a19-9d5d-96f2f321c6cf" />
<img width="1725" height="796" alt="PR 4026 Screenshot 4" src="https://github.com/user-attachments/assets/2777e508-ba1c-440a-a40a-06fb5e5a08c1" />

Commit:
Merge origin/development into veda-job-specific-template-changes

- Take Collaboration.jsx and yarn.lock from development
- Remove legacy Collaboration.css; keep rest of branch changes

Made-with: Cursor

Commit:
feat: open job-specific application template when clicking Collaboration cards

- Add react-router history navigation from Collaboration cards to /job-application
- Pass jobId, title, description, requirements, and category in location.state
- Remove legacy modal so job details are handled by JobApplicationForm page

Made-with: Cursor

Commit:
style: refine collaboration and job application layouts

- Collaboration: center container, responsive job grid spacing
- JobApplicationForm: balanced header bar (job title, Go, dropdown) attached to form

Made-with: Cursor

Anusha-Gali reviewed (changes requested):
Hi Veda,

Thanks for fixing the issue i had previously mentioned, however new issues popped up.
- The calendar in dark mode is in light mode
- Also the description is not visible anymore

<img width="1274" height="806" alt="Image" src="https://github.com/user-attachments/assets/e7b554ba-c729-4f8a-93b3-dba9d48950c9" />
<img width="1274" height="806" alt="Image" src="https://github.com/user-attachments/assets/749518f4-27be-49f2-bda7-184c6689481c" />

**Issues**
<img width="1274" height="806" alt="Image" src="https://github.com/user-attachments/assets/3b48883e-ccde-4c5f-b9e8-d2c581dec7ec" />
<img width="1274" height="806" alt="Image" src="https://github.com/user-attachments/assets/b6ae34a8-bce0-4132-a78b-33c7723b9e83" />
<img width="1274" height="806" alt="Image" src="https://github.com/user-attachments/assets/8e39d1c2-52a2-47d7-9f7c-a8ded988a82b" />

rohanrastogi311 reviewed (changes requested):
Hi Veda,

Job descriptions are not showing. Well done on previous fixes though! 

<img width="1728" height="1037" alt="Image" src="https://github.com/user-attachments/assets/f2c49a3e-06c3-4280-a356-eee98b738671" />
<img width="1728" height="1039" alt="Image" src="https://github.com/user-attachments/assets/25f1cd17-37df-4af3-b526-cd7105ca3027" />

rohanrastogi311 reviewed (changes requested):
Hi Veda, 

Job descriptions aren't showing. 

<img width="1727" height="1040" alt="Image" src="https://github.com/user-attachments/assets/3483fb6f-b670-4073-86e0-8398d6a89474" />

HemanthNidamanuru reviewed (changes requested):
Hi Veda,

The calendar is appearing in light mode even when the application is in dark mode. Additionally, the description is displayed incorrectly and the questions text is not visible in dark mode, and the Enter Job Title field is also appearing in light mode.

<img width="1898" height="1081" alt="Image" src="https://github.com/user-attachments/assets/670864f1-13b1-4801-bf2c-85ec14f0b2ac" />
<img width="1902" height="1090" alt="Image" src="https://github.com/user-attachments/assets/bfdd1340-0d1e-40cc-98a7-a7ae24f1fe8a" />
<img width="1901" height="1043" alt="Image" src="https://github.com/user-attachments/assets/8e3a3116-9a62-4f28-a866-7b57955b10f2" />

Commit:
Fix dark mode date icon and improve job description modal

Commit:
style: format job description modal JSX

HemanthNidamanuru reviewed (changes requested):
Hi Veda Charitha,

The calendar, description, and Enter Job Title field issues in dark mode are rectified. However, the questions are still repeating and the dropdown for question ("Will your volunteer time require documentation of your hours?") is not properly sized. Please check once

<img width="1171" height="92" alt="Image" src="https://github.com/user-attachments/assets/53544889-32d4-4247-88b0-b6ba9345d94c" />
<img width="1898" height="1032" alt="Image" src="https://github.com/user-attachments/assets/1d8ada91-1310-407f-86af-bf48c07243df" />
<img width="1898" height="1045" alt="Image" src="https://github.com/user-attachments/assets/fb27d0c5-256e-4a6f-92c9-78866f898e26" />

Anusha-Gali reviewed (changes requested):
Hi Veda,

Thanks for fixing the previous issues, however need your fixes on:
- The questions are repeating, due to which few questions do not have text box or the upload button, also Q8 on top and Q7 on bottom are not similar to the rest of the Questions UI.
- As you can see in the video attached, no matter which role i select, the form is redirected towards "Virtual Assistant".

<img width="1268" height="797" alt="Image" src="https://github.com/user-attachments/assets/31850503-9a7a-4204-96f2-d529e1d37303" />
<img width="1268" height="797" alt="Image" src="https://github.com/user-attachments/assets/65c92fbe-e004-4901-95f3-93d90a732585" />
<img width="1268" height="797" alt="Image" src="https://github.com/user-attachments/assets/c891c641-5641-461d-9db8-0362a7cefdfa" />

https://github.com/user-attachments/assets/ab9d0540-4222-4f9d-8132-b3d4f58a66ed

Commit:
Merge origin/development into veda-job-specific-template-changes

Resolve Collaboration.jsx: combine React hooks with useHistory; remove
obsolete fetchPositions that referenced removed setPositions.

Resolve Collaboration.module.css: keep branch card styling with modern
color notation.

Made-with: Cursor

Commit:
Fix job application form: match role from navigation, remove duplicate questions

Use location.state.jobTitle to select the correct form when coming from job
listings. Remove hardcoded Q1–8 that duplicated API questions and broke
answer indices. Normalize questionType (textbox, etc.), filter visible
questions, and align date/dropdown/textarea styling. Stylelint fixes.

Made-with: Cursor

Commit:
Job application: banner shows clicked role; pass jobTitle in URL

Use history push with search query so job title survives navigation. Add
bannerJobTitle for FORM FOR … heading; improve form title fuzzy matching.
Modal uses banner title for category and display.

Made-with: Cursor

Anusha-Gali reviewed (changes requested):
Hi Veda,

Thank you for fixing the previous errors, just one small UI issue - the numbers are visible twice in the form, do have a look at it. The rest of the functionality works as per requirement.

**Issue**
<img width="1264" height="742" alt="Image" src="https://github.com/user-attachments/assets/d202ddaa-5635-498a-8ba2-726140185932" />
<img width="1264" height="742" alt="Image" src="https://github.com/user-attachments/assets/6bc73d8c-9298-47b9-82b5-e80ddcdbe5f8" />

**Issues Fixed**
<img width="1264" height="742" alt="Image" src="https://github.com/user-attachments/assets/560cddba-b867-44e5-9598-4fed87b50a22" />
<img width="1264" height="742" alt="Image" src="https://github.com/user-attachments/assets/9f030ba9-ca3e-4173-b4d1-15d06a9e57bd" />
<img width="1264" height="742" alt="Image" src="https://github.com/user-attachments/assets/36553ac5-46f7-4892-8182-3bbaf70a93bf" />

Commit:
Prevent duplicate numbers on job application question labels

Strip embedded template enumeration (e.g. "1.)") when rendering numbered headings.

Made-with: Cursor

one-community commented:
Thank you all, merging!
